### PR TITLE
Ignore unknown properties and subscription api changes

### LINKED ISSUE #77: Support planned metadata on first and last event batch
See https://github.com/zalando/nakadi/pull/406

Unknown properties in the batch should be ignored, or at least the proposed `metadata` property needs to be recognized.

### PR DESCRIPTION
**Pull request by jhorstmann**
This fixes #77, #81 and #82

### CODE REVIEW

File: src/main/java/org/zalando/fahrschein/NakadiReader.java [hunk 1]
```diff
@@ -233,36 +248,56 @@ void runInternal() throws IOException, BackoffException {
...
+                            break;
+                        }
+                        case "metadata": {
+                            LOG.debug("Skipping metadata in event batch");
+                            jsonParser.nextToken();
+                            jsonParser.skipChildren();
+                            break;
+                        }
+                        default: {
+                            LOG.warn("Unexpected field [{}] in event batch", field);
```
Comment (lukasniemeier-zalando) [inline src/main/java/org/zalando/fahrschein/NakadiReader.java]: Is this just forwards-compatibility or do we expect this already today?

Comment (jhorstmann) [inline src/main/java/org/zalando/fahrschein/NakadiReader.java]: For compatibility mostly, also according to our api guidelines unknown fields should be ignored.

File: src/main/java/org/zalando/fahrschein/ManagedCursorManager.java [hunk 2]
```diff
@@ -24,49 +24,109 @@
...
         try (final ClientHttpResponse response = request.execute()) {
 
-            if (response.getStatusCode().value() == HttpStatus.NO_CONTENT.value()) {
-                LOG.warn("Cursor for event [{}] in partition [{}] with offset [{}] was already committed", eventName, cursor.getPartition(), cursor.getOffset());
-            } else if (response.getStatusCode().is2xxSuccessful()) {
-                LOG.debug("Successfully committed cursor for event [{}] in partition [{}] with offset [{}]", eventName, cursor.getPartition(), cursor.getOffset());
+            final int status = response.getStatusCode().value();
+            if (status == 204) {
+                LOG.debug("Successfully committed cursor for subscription [{}] to event [{}] in partition [{}] with offset [{}]", subscriptionId, eventName, cursor.getPartition(), cursor.getOffset());
+            } else if (status == 200) {
```
Comment (lukasniemeier-zalando) [inline src/main/java/org/zalando/fahrschein/ManagedCursorManager.java]: no more `HttpStatus.XXX`?

File: src/main/java/org/zalando/fahrschein/ManagedCursorManager.java [hunk 3]
```diff
@@ -24,49 +24,109 @@
...
+                LOG.warn("Cursor for subscription [{}] to event [{}] in partition [{}] with offset [{}] was already committed", subscriptionId, eventName, cursor.getPartition(), cursor.getOffset());
+            } else {
+                LOG.warn("Unexpected status code [{}] for subscription [{}] to event [{}] in partition [{}] with offset [{}]", status, subscriptionId, eventName, cursor.getPartition(), cursor.getOffset());
```
Comment (lukasniemeier-zalando) [inline src/main/java/org/zalando/fahrschein/ManagedCursorManager.java]: Can one safely continue in case the cursors is not committed?

Comment (jhorstmann) [inline src/main/java/org/zalando/fahrschein/ManagedCursorManager.java]: It should still be a succesful response, codes >= 400 are handled by `ProblemHandlingClientHttpRequest`. Not sure what would happen with the 3xx range though.

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**